developer_guide.md

```diff
@@ -92,12 +92,61 @@ cd backend/src/ && go test ./...
 ### Frontend
 TODO: add instruction
 
-### DSL
+### Python SDK
 ```bash
-pip install ./dsl/ --upgrade && python ./dsl/tests/main.py
-pip install ./dsl-compiler/ --upgrade && python ./dsl-compiler/tests/main.py
+python -m venv /tmp/.env && source /tmp/.env/bin/activate # Optional, stops you installing dependencies globally.
```

Comment:
I think we are suggesting miniconda in sdk install documentation, so we can just assume developer already have a virtual env, or make it an orthogonal section

Reply:
No problems, I can't seem to find that documentation would you mind linking me to it so I can add the link in the code? I'm not seeing any mentions of miniconda in this repo.

Reply:
It's mentioned in kubeflow/website doc repo: https://www.kubeflow.org/docs/pipelines/sdk/install-sdk/